src/components/FullscreenLoadingIndicator.js

```diff
@@ -25,18 +34,48 @@ const defaultProps = {
  * @param {Object} props
  * @returns {JSX.Element}
  */
-const FullScreenLoadingIndicator = (props) => {
-    if (!props.visible) {
-        return null;
+class FullScreenLoadingIndicator extends React.Component {
```

Comment:
Let's add a comment here telling why do we need the class component here so.

Reply:
Other class components do not explain why a class component is used. Why is it any different here?

Reply:
Yeah, just checked the style guide again. we are fine here. Point was that this component does not have state or bound methods so it is not apparent that it needs to be a class component. So leaving a comment will prevent others from thinking to change this. But it's fine code says it all. we need lifecycle methods that are clearly visible here.